### ISSUE TITLE
java.lang.NullPointerException on Android 12

### ISSUE THREAD

**Issue by jinkaiYu** [OneSignal/OneSignal-Android-SDK#1550]
![image](https://user-images.githubusercontent.com/18320608/158962893-46a46c91-cd66-4147-a427-5b62c615b530.png)

**Comment by jinkaiYu** [comment]
SDK version 4.6.3

**Comment by nan-li** [comment]
Hi @jinkaiYu,

Thank you for reporting. There is not much information in the screenshot you attached.

Can you tell me more about this exception? 

Do you have a stack trace since you say that it is a `NullPointerException`?

**Comment by jinkaiYu** [comment]
![image](https://user-images.githubusercontent.com/18320608/159123250-bfecefd5-b4cb-48ce-82b4-1150162acd02.png)

**Comment by jinkaiYu** [comment]
more info about it.

**Comment by MHPtweidinger** [comment]
the crash also occurred for us. but it doesn't seem to be related to android 12. 
in our case it happened on a device running android 9. 

 ```
java.lang.NullPointerException: Attempt to invoke virtual method 'void com.onesignal.s1.q()' on a null object reference
	 at com.onesignal.OneSignal.init(OneSignal.java:858)
	 at com.onesignal.OneSignal.setAppId(OneSignal.java:713)
	 at com.onesignal.OneSignal.reassignDelayedInitParams(OneSignal.java:1158)
	 at com.onesignal.OneSignal.onRemoteParamSet(OneSignal.java:865)
	 at com.onesignal.OneSignal$6.complete(OneSignal.java:1099)
	 at com.onesignal.OneSignalRemoteParams.processJson(OneSignalRemoteParams.java:206)
	 at com.onesignal.OneSignalRemoteParams.access$100(OneSignalRemoteParams.java:12)
	 at com.onesignal.OneSignalRemoteParams$1.onSuccess(OneSignalRemoteParams.java:151)
	 at com.onesignal.OneSignalRestClient$5.run(OneSignalRestClient.java:292)
```

**Comment by nan-li** [comment]
Hi again @jinkaiYu, do you have the **full** error for the `NullPointerException` that shows exactly which method and object it is? Something like this:

> Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.onesignal.OSOutcomeEventsController.sendSavedOutcomes()' on a null object reference
at com.onesignal.OneSignal.init(OneSignal.java:837)
at com.onesignal.OneSignal.setAppId(OneSignal.java:692)
...

Hi @MHPtweidinger,
Your code is obfuscated but the stack trace looks like it may be related to #1514 instead of this issue.

**Comment by jinkaiYu** [comment]
@nan-li Thanks, I would like to know in which version will this issue be fixed?

**Comment by jinkaiYu** [comment]
@nan-li Hello, I want to ask another question. We need to display In App Message on a specified page (Home page). Currently, the cost of using PauseInAppMessage(boolean) is too high. All pages need to call this. Is there any plan to let developers designate a fixed page to pop up InAppMessage?

**Comment by jkasten2** [comment]
> @nan-li Thanks, I would like to know in which version will this issue be fixed?

@jinkaiYu This was fixed in [4.7.1 Release](https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/4.7.1)


> We need to display In App Message on a specified page (Home page). Currently, the cost of using PauseInAppMessage(boolean) is too high. All pages need to call this. Is there any plan to let developers designate a fixed page to pop up InAppMessage?

Please create a new issue for this one with more details. I don't understand what you mean by the cost is too high. How many different Activities do you have? Is there some kind of specific property you are looking to filter on?
